components/proxy/src/main/java/com/hotels/styx/admin/dashboard/DashboardData.java

```diff
@@ -206,6 +225,13 @@ private Backend(BackendService application) {
             this.responsesSupplier = new ResponseCodeSupplier(metrics, METER, prefix, true);
         }
 
+        void unregister() {
+            registeredOrigins.forEach(origin -> {
```

Comment:
Why do we have a new field `registeredOrigins` that contains the same list as the existing field `List<Origin> origin`?

Reply:
This is so that we unregister only those origins that actually have been registered.
I noticed that in some test cases (DashboardDataSupplier) the `origins` list contained different origins to those from `registeredOrigins`. But this could have been just a unusual or unrealistic dependency injection in the test itself.